client/app.jsx

```diff
@@ -36,7 +53,32 @@ export default class App extends React.Component {
     this.setState({ watchlist: entries });
   }
 
+  OpenMenu() {
+    if (this.state.menuOpen === true) {
+      this.setState({ menuOpen: false });
+    } else {
+      this.setState({ menuOpen: true });
+    }
+  }
+
+  OpenWatchlist() {
+    this.setState({ watchlistOpen: true });
+  }
+
+  GoHome() {
+    this.setState({ watchlistOpen: false });
+  }
+
   render() {
-    return <Home text="TV Diary" SetSearchResults= { this.SetSearchResults } searchResults = { this.state.searchResults } watchlist = {this.state.watchlist} AddToWatchlist = {this.AddToWatchlist}/>;
+    if (this.state.watchlistOpen === false) {
+      return <div>
+        <Home text="TV Diary" SetSearchResults={this.SetSearchResults} searchResults={this.state.searchResults} watchlist={this.state.watchlist}
+          AddToWatchlist={this.AddToWatchlist} menu={this.OpenMenu} menuOpen={this.state.menuOpen} OpenWatchlist={this.OpenWatchlist}
+          isWatchlistOpen={this.state.watchlistOpen} GoHome = {this.GoHome} />
+      </div>;
+    } else {
+      return <Watchlist menu={this.OpenMenu} menuOpen={this.state.menuOpen === false} GoHome={this.GoHome} OpenWatchlist={this.OpenWatchlist}
+        isWatchlistOpen={this.state.watchlistOpen} watchlist={this.state.watchlist} />;
+    }
```

Comment:
Your app drawer should be in here instead of in every single screen. That should help remove a lot of duplication in your code.